pkg/target/target_test.go

```diff
@@ -693,13 +695,74 @@ func TestMatcher_Match(t *testing.T) {
 	}
 }
 
+func TestNamespaceCache(t *testing.T) {
+	ns1 := makeNamespace("my-ns1", map[string]string{"ns1": "label"})
+	ns2 := makeNamespace("my-ns2", map[string]string{"ns2": "label"})
+	tests := []struct {
+		name    string
+		addNs   map[string]interface{}
+		wantNs  map[string]*corev1.Namespace
+		wantErr bool
```

Comment:
Use `wantErr error` and `errors.Is` to validate it. Create errors with `errors.New` as needed.